### DISCUSSION TITLE
Find Many with the startkey and endkey parameters.

### DISCUSSION THREAD

**Discussion by hamiltonsalazar** [apache/couchdb#3008] [Q&A] [answered]
I have a db and i want to use this view:
```
function (doc) {
    if (doc.sales) {
        for (var sale in doc.sales) {
            if (doc.sales[sale].products) {
                for (var product in doc.sales[sale].products) {
                    emit([doc.sales[sale].products[product].name, doc._id], null);
                }
            }
        }
    }
}
```

I'm using the starkey and endkey parameters as in the documentation in [Views Collation/Examples](https://docs.couchdb.org/en/stable/ddocs/views/collation.html#examples) where it uses `startkey=["XYZ"]&endkey=["XYZ", {}]`

If i send `curl "http://anna:secret@localhost:5984/pruebaproyecto/_design/viewsales2/_view/sales_products_count?group=true&startkey=\[\"Almohadilla Dactilar Artline X 1 Unidad Huellero 5000 Imps\"\]&endkey=\[\"Almohadilla Dactilar Artline X 1 Unidad Huellero 5000 Imps\",\{\}\]"`
CouchDB responds with: 
> curl: (52) Empty reply from server

But if i change the order in the emit function:
```
function (doc) {
    if (doc.sales) {
        for (var sale in doc.sales) {
            if (doc.sales[sale].products) {
                for (var product in doc.sales[sale].products) {
                    emit([doc._id, doc.sales[sale].products[product].name], null);
                }
            }
        }
    }
}
```
And i send ` curl "http://anna:secret@localhost:5984/pruebaproyecto/_design/viewsales2/_view/sales_products_count?group=true&startkey=\[\"1\"\]&endkey=\[\"1\",\{\}\]"` , CouchDB worked and responds with:
>{"rows":[
{"key":["1","Almohadilla Dactilar Artline X 1 Unidad Huellero 5000 Imps"],"value":20},
{"key":["1","Arbol Espuma Esc 1:200 X5 Mires"],"value":9},
{"key":["1","Arbusto Esc 1:50 X 5 Mires"],"value":20},
{"key":["1","Aromatica Jaibel Manzanilla X 25 Unidades"],"value":4},
{"key":["1","Aromatica Jaibel Tradicional Limonaria X 23 Unidades"],"value":21},
{"key":["1","Aromatica Jaibel Tradicional Manzanilla X 23 Unidades"],"value":20},
{"key":["1","Aserrin Verde Mires"],"value":19},
{"key":["1","Barquillos Flaquito Nevado Avellanas Y Chocolate Oscuro X 4"],"value":20},
{"key":["1","Block Amarillo Anotaciones 1/2 Cartacuadriculado 50h Norma"],"value":13},
{"key":["1","Block Argollado Pappyer Amarillo Carta Cuadriculado"],"value":21},
{"key":["1","BLOCK BOND CARTA CUADROS X 80H. IM├üGENES"],"value":19},
{"key":["1","BLOCK BOND OFICIO RAYADO X 80H"],"value":16},
{"key":["1","BLOCK BOND OFICIO RAYADO X 80H. IM├üGENES"],"value":22},
{"key":["1","Block Edad Media 1/4 20h Nessan"],"value":16},
{"key":["1","Block Edad Media 1/8 20h Nessan"],"value":20},
{"key":["1","Block Iris Carta 40 H Primavera"],"value":11},
{"key":["1","Block Iris Carta Nacional 20h Nessan"],"value":20},
{"key":["1","Bolig. Roller Fashion Week E Monza Naranja Ray Inoxcrom"],"value":21},
{"key":["1","Bol├¡grafo / Lapicero Ref 062 Negro - Unidad a $900"],"value":6},
{"key":["1","Borrador Miga De Pan Mp20 X 2 Pelikan"],"value":9},
{"key":["1","Borrador Negro Miga De Pan Studmark X 3 Unidades"],"value":19},
{"key":["1","Calculadora Fx9860 Graficadora Gll Casio Memoria Sd"],"value":9},
{"key":["1","Calculadora Gratificadora Casio Fx-cg50 Nueva"],"value":6},
{"key":["1","Candy Ventilador Pricesas Disney"],"value":8},
{"key":["1","Carpeta Legajadora Oficio C/g Naranja Fluorescente K009703 Keepermate"],"value":8},
{"key":["1","Carpeta Plastica Legajadora Tama├▒o Oficio / Keepermate"],"value":9},
{"key":["1","Cereal Karymba Bolsa 120gr Chocolate"],"value":13},
{"key":["1","Chinchon Pl├ístico Colores X 100 St-04056 Studmark"],"value":20},
{"key":["1","Cinta Ancha Transparente 48 X 40 Tesa"],"value":17},
{"key":["1","Cintas Con Silicona Y Pegante En Barra Tesa"],"value":20},
{"key":["1","Colores Schwan Art X24 Stabilo"],"value":14},
{"key":["1","Colores Schwan Art X48 Stabilo"],"value":19},
{"key":["1","Cuaderno Musica Grapado Peque├▒a Nessan"],"value":17},
{"key":["1","Curvigrafo Faber Castell"],"value":13},
{"key":["1","Curvigrafo Mediano Faber Castell"],"value":20},
{"key":["1","Diamantina Con Pegamento 60ml Pelikan Magneta"],"value":8},
{"key":["1","Diamantina Con Pegamento 60ml Pelikan Verde"],"value":8},
{"key":["1","Escalimetro De Abanico Pointer X 16"],"value":20},
{"key":["1","Estilografo Parker Sonnet Black Laca C8637 Dorado"],"value":21},
{"key":["1","Fichas Memotecnica Rayado 10 X 15 Cm.-50h Nessan"],"value":17},
{"key":["1","Galletas Integrales Avena Y Sabor A Frutos Rojos 156gr X 6"],"value":18},
{"key":["1","Galletas Mediterraneas Con Trozos De Almendras Kurabie 80 Gr"],"value":3},
{"key":["1","Gancho Clip Jumbo X 50 Triton"],"value":9},
{"key":["1","Gancho Clip Mariposa No. 2 X 50 Wingo"],"value":13},
{"key":["1","Gancho Cosedora Galvanizado 26/6 X 5000 Triton x 50 unidades"],"value":20},
{"key":["1","Gancho Lotero 1 X 12 Triton"],"value":20},
{"key":["1","Gomas Happy Starmix 80gr"],"value":9},
{"key":["1","Gu├¡a Celuguia Norma Color Amarillo"],"value":15},
{"key":["1","Hersheys Milk Chocolate 43gr. X 36 Unidades"],"value":13},
{"key":["1","Hilo Nylon 0.7 Crystal Tec"],"value":13},
{"key":["1","Iman Circulo 3 Cm Mires"],"value":4},
{"key":["1","Impresi├│n De Fotograf├¡as 10x15cm 10x10cm"],"value":3},
{"key":["1","Jaibel Te Negro Clasico 20 Und 36g"],"value":20},
{"key":["1","Juego Didactico Tangram Mediano"],"value":15},
{"key":["1","Juego Did├íctico Tangram Peque├▒o En Madera 16 X 16 Cm"],"value":2},
{"key":["1","Kit Masa Mold. Slime Magic Diy Lab Hazlo Tu Mismo Mk138"],"value":4},
{"key":["1","Lapiz Castell 9000 2h X 3 Unid. Faber Castell"],"value":14},
{"key":["1","L├ípiz Corrector Liquido Stabilo"],"value":19},
{"key":["1","Lapiz Corrector Mini 3.5 Ml Liquid Paper"],"value":15},
{"key":["1","Lapiz De Dibujo Othello X 12 Stabilo"],"value":13},
{"key":["1","Lapiz Hb No. 2 Papermate"],"value":15},
{"key":["1","Maqueta Teja Escala 1:20 X 100 Mires"],"value":19},
{"key":["1","M├íquina Anilladora Encuadernadora Doble O Met├ílica Akiles"],"value":17},
{"key":["1","Marcador Para Tela Morado O Violeta E-4500"],"value":17},
{"key":["1","Marcadores Edding Color Happy Box Ch20+1 Rotuladores"],"value":9},
{"key":["1","Micropunta Point 88 Stabilo X 8 Colores Pasteles"],"value":1},
{"key":["1","Micropunta Point 88 X 5 Neon Stabilo"],"value":20},
{"key":["1","Minas Portaminas 0.7 Hb X 2 Unid. Faber Castell"],"value":6},
{"key":["1","Minifacturero Vert. periodico 1/64 X 40h X12 Fabrifolder"],"value":12},
{"key":["1","Monedas De Chocolate Caja X12 Bolsas 45 De Unidades"],"value":20},
{"key":["1","Morral Juvenil + Cartuchera Estampados Minnie 1911-1912"],"value":9},
{"key":["1","Mouse ├ôptico Inal├ímbrico Logitech G903 Lightspeed Gamer"],"value":9},
{"key":["1","Palmera Esc 1:50 X 6 Mires"],"value":4},
{"key":["1","Palo De Balso Redondo 10 - Balco"],"value":18},
{"key":["1","Pegante 250 Gr. Papel Si Pega"],"value":15},
{"key":["1","Pegante En Barra 22 Gr. Stick Tesa"],"value":4},
{"key":["1","Pegante En Barra 40 Gr. Stick Tesa"],"value":22},
{"key":["1","Pegante Papel 225 Ml. Elmer`s"],"value":20},
{"key":["1","Pincel Cerda Redondo 582 No. 2"],"value":2},
{"key":["1","Pintura Al Frio Durazno 06 Franco Arte"],"value":20},
{"key":["1","Plumigrafo Point 88 Mini Stabilo"],"value":6},
{"key":["1","Plumigrafo Point 88 Stabilo"],"value":5},
{"key":["1","Plumigrafo Point 88 Stabilo - Unid - Unidad A $3000"],"value":20},
{"key":["1","Plumigrafo Point 88 X25 Stabilo"],"value":8},
{"key":["1","Plumillas Para Tinta China Botero"],"value":4},
{"key":["1","Regla T Escolar 30 Cm. 457 Faber Castell"],"value":20},
{"key":["1","Repuesto Para Folder Tama├▒o Mediano 95 Cuadriculado Norma"],"value":8},
{"key":["1","Repuesto Visturi Grande Caja X 10 Unidades"],"value":6},
{"key":["1","Resaltador Colores Pastel Boss Original Stabilo"],"value":6},
{"key":["1","Resaltador Colores Pastel X6 Original Boss Stabilo"],"value":12},
{"key":["1","Resaltador Flash X4 Stabilo  "],"value":15},
{"key":["1","Rodillo De Entrenamiento Para Bicicletas"],"value":9},
{"key":["1","Roller Balloon Naranja 311 Lamy"],"value":20},
{"key":["1","Roller Parker 5 Th C8782 Distric Urban Premium Ebony"],"value":20},
{"key":["1","Rotulo Adhes. Blanco 2008 Dimatic"],"value":21},
{"key":["1","Rotulo Adhes. Blanco 8153 Dimatic"],"value":19},
{"key":["1","Rotulo Adhes. Blanco 8642 Dimatic"],"value":6},
{"key":["1","Rotulo Adhes. Redondo 13 Plata. Dimatic"],"value":20},
{"key":["1","Rotulo Adhesivo Blanco 10053 Dimatic"],"value":12},
{"key":["1","Sacaganchos Rf 2600 Triton"],"value":8},
{"key":["1","Separador 105 X 5 Amarillo Fabrifolder"],"value":21},
{"key":["1","Set Plum├│n Y Plumigrafo Stabilo - Unidad A $3250"],"value":9},
{"key":["1","Silicona Liquida 100 Ml Tesa"],"value":14},
{"key":["1","Silicona Liquida 60 Ml Faber Castell"],"value":20},
{"key":["1","Sobre Manila 1/2 Oficio 17.5 X 24 Norma"],"value":15},
{"key":["1","Tabla Legajadora Oficio En Madera Fabrifolder Mdf"],"value":7},
{"key":["1","Tablero Acrilico Con Caballete Artecma"],"value":8},
{"key":["1","Tajalapiz Met├ílico Doble Servicio Keyroad"],"value":7},
{"key":["1","Tajalapiz Star Doble Keyroad"],"value":20},
{"key":["1","TALLER DE LETRAS"],"value":12},
{"key":["1","Texto Constituci├│n Pol├¡tica De Colombia 1991"],"value":6},
{"key":["1","Tijera Oficina 7 Pulgadas Talbot"],"value":5},
{"key":["1","Tijeras Escolar 5 Triton"],"value":13},
{"key":["1","Tinta Hp Cz105al Negro No. 662xl"],"value":21},
{"key":["1","Tiza Pastel Chalk X 12 Pushing Corriente"],"value":20},
{"key":["1","Twix Galleta Y Chocolate 50gr X 36 Uds - kg a $2"],"value":8},
{"key":["1","Velita Escarchada N┬║ 0 Sempertex"],"value":14}
]}

**Why i can't get the results of view where the key include a `doc.sales[sale].products[product].name` specific and any `doc._id`?**

**Comment by janl** [comment]
what error are you getting in your couch.log when you get `curl: (52) Empty reply from server` and what else do you get when you add `-v` to your `curl` command?

**Reply by hamiltonsalazar** [reply]
When i add `-v` to my `curl` commnad: 

> *   Trying ::1...
> * TCP_NODELAY set
> *   Trying 127.0.0.1...
> * TCP_NODELAY set
> * Connected to localhost (127.0.0.1) port 5984 (#0)
> * Server auth using Basic with user 'anna'
> GET /pruebaproyecto/_design/viewsales2/_view/sales_products_count?group=true&startkey=["Almohadilla Dactilar Artline X 1 Unidad Huellero 5000 Imps"]&endkey=["Almohadilla Dactilar Artline X 1 Unidad Huellero 5000 Imps",{}] HTTP/1.1
> Host: localhost:5984
> Authorization: Basic YW5uYTpzZWNyZXQ=
> User-Agent: curl/7.55.1
> Accept: */*
>
> * Empty reply from server
> * Connection #0 to host localhost left intact
curl: (52) Empty reply from server

I don't know how to see my couch.log on Windows

**Comment by eiri** [comment, marked as answer]
I suspect this is because you have spaces in your product names and they got misinterpreted during http request or parsing. Try to urlescape your keys, something like `curl "http://anna:secret@localhost:5984/pruebaproyecto/_design/viewsales2/_view/sales_products_count?group=true&startkey=%5B%22Almohadilla%20Dactilar%20Artline%20X%201%20Unidad%20Huellero%205000%20Imps%22%5D&endkey=%5B%22Almohadilla%20Dactilar%20Artline%20X%201%20Unidad%20Huellero%205000%20Imps%22%2C%7B%7D%5D"`
